OpenDreamShared/Compiler/DMPreprocessor/DMPreprocessorLexer.cs

```diff
@@ -197,15 +197,56 @@ class DMPreprocessorLexer : TextLexer {
                         textBuilder.Append('@');
                         textBuilder.Append(delimiter);
 
-                        do {
-                            c = Advance();
+                        if (delimiter == '{')
+                        {
+                            delimiter = '}';
+                            char prev;
+                            do
+                            {
+                                prev = c;
+                                c = Advance();
+
+                                textBuilder.Append(c);
+                                if (c == '}' && prev == '\"')
+                                {
+                                    break;
+                                }
+                            } while (!AtEndOfSource);
+
+                            if (AtEndOfSource)
+                            {
+                                token = CreateToken(TokenType.Error, "Unterminated raw string");
+                            }
+                        }
+                        else
+                        {
+                            do
+                            {
+                                c = Advance();
+                                textBuilder.Append(c);
+                            } while (c != delimiter);
+                        }
 
-                            textBuilder.Append(c);
-                        } while (c != delimiter && c != '\n');
                         Advance();
 
+                        var startIdx = 2;
+                        var stopLength = 3;
+                        if (delimiter == '}')
+                        {
+                            if (textBuilder[2] == '"')
+                            {
+                                startIdx++;
+                                stopLength = startIdx + 1;
+                            }
+
+                            if (textBuilder[textBuilder.Length-2] == '"')
+                            {
+                                stopLength = startIdx + 2;
+                            }
+                        }
```

Comment:
It'd be better to instead have two separate `CreateToken` calls after building the text for each type of raw string.
`token = CreateToken(TokenType.DM_Preproc_ConstantString, text, text.Substring(2, text.Length - 3));` after line 227, and
`token = CreateToken(TokenType.DM_Preproc_ConstantString, text, text.Substring(3, text.Length - 5));` after line 219.